### ISSUE TITLE
bug(MatSlideToggle): Unlabelled slide toggle cannot be labeled by a dynamically set aria-label

### ISSUE THREAD

**Issue by alicequiros** [angular/components#23094]
#### Reproduction

https://angular-j8a87w.stackblitz.io

Steps to reproduce:
1. Open Chrome inspector and inspect the mat-slide-toggle, note that it doesn't have an aria-label
2. Expand the DOM tree and note that the nested input also doesn't have an aria-label
3. Inspect the checkbox on the page, which is similarly labeled, and note that the aria-label applies properly
 
![image](https://user-images.githubusercontent.com/81990996/124002222-7887e280-d9a3-11eb-86f0-488f68e1cae3.png)
 
#### Expected Behavior

[attr.aria-label]="..." should apply an aria-label to the slide toggle and the nested input element

#### Actual Behavior

Applying a dynamic aria-label to an unlabeled mat-slide-toggle does not currently seem possible

#### Environment

- Angular: Stackblitz was copied from the documentation at https://material.angular.io/components/slide-toggle
- Browser(s): Chrome Version 89.0.4389.114

**Comment by crisbeto** [comment]
The `attr.aria-label` won't work because we don't have a good way of picking it up. We do have an `aria-label` input so doing something like `[aria-label]="..."` should work.

**Comment by alicequiros** [comment]
Ah I see. Are there any plans to support ```aria-describedby``` in a similar way to ```aria-label```?

**Comment by crisbeto** [comment]
I've submitted #23104 to add support for it.
